pkg/capture/capture_manager.go

```diff
@@ -192,17 +220,30 @@ func (cm *Manager) Status(ctx context.Context, ifaces ...string) (statusmap map[
 }
 
 // Update the configuration for all (or a set of) interfaces
-func (cm *Manager) Update(ctx context.Context, ifaces config.Ifaces) {
+func (cm *Manager) Update(ctx context.Context, ifaces config.Ifaces) (enabled, updated, disabled []string, err error) {
+	// Validate the config before doing anything else
+	err = ifaces.Validate()
+	if err != nil {
+		return
+	}
 
 	// Build set of interfaces to enable / disable
 	var (
-		ifaceSet                    = make(map[string]struct{})
-		enableIfaces, disableIfaces []string
+		ifaceSet                                  = make(map[string]struct{})
+		enableIfaces, updateIfaces, disableIfaces []string
 	)
-	for iface := range ifaces {
+	for iface, cfg := range ifaces {
 		ifaceSet[iface] = struct{}{}
 		if _, exists := cm.captures.Get(iface); !exists {
 			enableIfaces = append(enableIfaces, iface)
+		} else {
+			updatedCfg := cfg
+			runtimeCfg := cm.lastAppliedConfig[iface]
```

Comment:
Well spotted. I'll wrap this section with a global lock then.

Reply:
Fair warning: That might create another deadlock (at least IIRC I deliberately split the non-blocking (`Update()`) and blocking (`update()`) part because the locking part is used from `Close()` also). Might be that by adding a lock here things fall apart again, but I can't say for sure...